app/controllers/news_items_controller.rb

```diff
@@ -1,9 +1,17 @@
 class NewsItemsController < ApplicationController
   def index
-    @news_items = NewsItem.published.to_a
+    @news_items = news_item_scope.to_a
   end
 
   def show
-    @news_item = NewsItem.published.find(params[:id])
+    @news_item = news_item_scope.where(slug: params[:slug]).first ||
+                 news_item_scope.where(id: params[:id]).first ||
+                 raise(ActiveRecord::RecordNotFound)
```

Comment:
Gotcha, we probably should do a redirect to the correct URL, but let's do that as a separate PR.